src/app-layer-ssl.c

```diff
@@ -897,9 +897,13 @@ static inline int TLSDecodeHSHelloCipherSuites(SSLState *ssl_state,
         }
 
         if (enable_ja3) {
-            int rc = Ja3BufferAppendBuffer(&ssl_state->curr_connp->ja3_str, &ja3_cipher_suites);
-            if (rc == -1) {
-                return -1;
+            if (ssl_state->curr_connp->ja3_hash == NULL) {
```

Comment:
can't we check this earlier? Like when we init enable_ja3? Seems wasteful to init ja3_cipher_suites if we can already know it's not going to be used. Also don't like how we have different places we need to free it this way.

Reply:
Indeed, nice :-)